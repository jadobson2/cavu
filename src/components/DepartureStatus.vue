<template>
  <div
    class="departure-status inline-flex w-fit rounded-l-lg rounded-r-3xl"
    :class="`departure-status--${statusSlug}`"
    data-test="departure-status"
  >
    <span class="block ml-3 pl-4 pr-7 py-2 rounded-l-lg rounded-r-3xl bg-white">
    {{ statusText }}
  </span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { DepartureStatusCode } from '@/types'
import { getStatus } from '@/utils/departures'
import { slugify } from '@/utils/string'

export default Vue.extend({
  name: 'DepartureStatus',
  props: {
    statusText: String
  },
  computed: {
    status (): DepartureStatusCode {
      return getStatus(this.statusText)
    },
    statusSlug (): string {
      return slugify(this.status)
    }
  }
})
</script>

<style lang="scss" scoped>
.departure-status {
  $statuses: (
    boarding: (theme('colors.green'), theme('colors.green')),
    cancelled: (theme('colors.red'), theme('colors.red')),
    delayed: (theme('colors.red'), theme('colors.red')),
    departed: (theme('colors.green'), theme('colors.green')),
    gate-closed: (theme('colors.yellow.DEFAULT'), theme('colors.greyscale.90')),
    gate-closing: (theme('colors.orange'), theme('colors.orange')),
    gate-open: (theme('colors.blue'), theme('colors.greyscale.90')),
  );

  @each $status, $values in $statuses {
    &--#{$status} {
      background-color: nth($values, 1);
      color: nth($values, 2);
    }
  }
}
</style>
