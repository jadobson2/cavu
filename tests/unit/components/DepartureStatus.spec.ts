import { shallowMount } from '@vue/test-utils'
import DepartureStatus from '@/components/DepartureStatus.vue'

describe('DepartureStatus', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(DepartureStatus, {
      propsData: { statusText: 'Departed 10:45' }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
