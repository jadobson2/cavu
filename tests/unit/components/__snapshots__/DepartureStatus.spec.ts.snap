// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepartureStatus renders correctly 1`] = `
<div
  class="departure-status inline-flex w-fit rounded-l-lg rounded-r-3xl departure-status--departed"
  data-test="departure-status"
>
  <span
    class="block ml-3 pl-4 pr-7 py-2 rounded-l-lg rounded-r-3xl bg-white"
  >
    
  Departed 10:45

  </span>
</div>
`;
